staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go

```diff
@@ -123,3 +128,55 @@ func TestSelectionPredicate(t *testing.T) {
 		}
 	}
 }
+
+func TestSelectionPredicateMatcherIndex(t *testing.T) {
+	testCases := map[string]struct {
+		labelSelector, fieldSelector string
+		indexLabels                  []string
+		indexFields                  []string
+		excepted                     []MatchValue
+	}{
+		"A": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{"bar"},
+			indexFields:   []string{},
+			excepted:      nil,
+		},
+		"B": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{},
+			indexFields:   []string{"uid"},
+			excepted:      []MatchValue{{IndexName: "f:uid", Value: "12345"}},
+		},
```

Comment:
nit: better to have another test case with matching label, but no matching field

Reply:
> nit: better to have another test case with matching label, but no matching field

done